### PR TITLE
Removes the Lagrange form of the SRS

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

As discussed in slack -- the lagrange form of the SRS is no longer needed due to the advancements made in R&D. 

This PR removes the lagrange form of the SRS.

Changes:

- First commit, I just grepped the codebase for lagrange being used with pippenger and removed it

- Second commit; I did the IFFT of the wire polynomials before committing to them, which made their monomial form available to commit to. The monomial form can be fetched from the polynomial cache by using the `wire_tag` without the lagrange postfix.

# Checklist:

- [ ] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [ ] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [ ] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/aztec/CMakeLists.txt [hunk 1]
```diff
@@ -2,12 +2,12 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
 # Enable the following warnings project wide.
-# If any compilation issues arise in the future, they should not be silenced here but rather in the
```
Comment (kevaundray) [inline cpp/src/aztec/CMakeLists.txt:5]: seems my formatter removed the extra whitespace here on saving

File: cpp/src/aztec/CMakeLists.txt [hunk 2]
```diff
@@ -2,12 +2,12 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
 # Enable the following warnings project wide.
-# If any compilation issues arise in the future, they should not be silenced here but rather in the 
+# If any compilation issues arise in the future, they should not be silenced here but rather in the
 # module's own CMakeLists.txt by adding conditional compilation flags like the following
 # ```
-#   if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-#     target_compile_options(crypto_blake3s_full_objects PRIVATE -Wno-error=shorten-64-to-32)
```
Comment (kevaundray) [inline cpp/src/aztec/CMakeLists.txt:9]: It did the same thing here, though this one seems undeseriable

File: cpp/src/aztec/proof_system/work_queue/work_queue.cpp [hunk 3]
```diff
@@ -58,25 +58,6 @@ size_t work_queue::get_scalar_multiplication_size(const size_t work_item_number)
     return 0;
 }
 
-/**
- * Returns a boolean that denotes if the scalar multiplication is to be done with the Monomial SRS or Lagrange SRS.
```
Comment (kevaundray) [inline cpp/src/aztec/proof_system/work_queue/work_queue.cpp:62]: Removed this as the scalar multiplication type will always be monomial SRS

Comment (kevaundray) [reply cpp/src/aztec/proof_system/work_queue/work_queue.cpp:62]: If you want to maintain backwards compatibility/ some deprecation policy -- I can leave this in and have it always return false

**Comment by kevaundray** [comment]
(I can't compile locally so I'm using the CI as a way to compile -- should be fine as it gets squashed :) )

File: cpp/src/aztec/proof_system/work_queue/work_queue.cpp (hunk 3 above)
Comment (suyash67) [reply cpp/src/aztec/proof_system/work_queue/work_queue.cpp:62]: Better to remove if we don't use it. So I agree with you.

Comment (kevaundray) [reply cpp/src/aztec/proof_system/work_queue/work_queue.cpp:62]: sounds good

**Review by suyash67 — APPROVED** [review]
Looks good to me, thanks a lot @kevaundray

**Pull request merged**